house/grid.py

```diff
@@ -32,6 +35,28 @@ def get_neighbours(self) -> list:
         
         return nodes
 
+
+    #Check if the GridNode has neighbours to its diagonals in the same plane
+    def has_diagonal_neighbours(self):
+        for dx, dy, dz in [(-1, 0, -1), (-1, 0, 1),(1, 0, 1), (1, 0, -1)]:
```

Comment:
Use constants defined in directions for easy legibility. Could just create a list of diagonals in directions.py and say `for dx, dy, dz in diagonals`